src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -294,7 +294,7 @@ public function get($id, $invalidBehavior = /* self::EXCEPTION_ON_INVALID_REFERE
             }
 
             if (isset($this->loading[$id])) {
-                throw new ServiceCircularReferenceException($id, array_keys($this->loading));
+                throw new ServiceCircularReferenceException($id, array_merge(array_keys($this->loading), array($id)));
```

Comment:
Loosely-related improvement of the exception message: displays `"foo -> bar -> foo"` instead of just `"foo -> bar"` (already done this way in other places that throw `ServiceCircularReferenceException`.)